src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -3294,6 +3299,320 @@ .locals init (C.<f>d__0 V_0)
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(1)").WithArguments("B1", "Create").WithLocation(22, 19),
+                // (23,19): error CS0656: Missing compiler required member 'B2.Task'
+                //     async T2 f2() => await Task.Delay(2);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(2)").WithArguments("B2", "Task").WithLocation(23, 19),
+                // (23,19): error CS0656: Missing compiler required member 'B2.Create'
+                //     async T2 f2() => await Task.Delay(2);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(2)").WithArguments("B2", "Create").WithLocation(23, 19),
+                // (24,19): error CS0656: Missing compiler required member 'B3.Task'
+                //     async T3 f3() => await Task.Delay(3);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(3)").WithArguments("B3", "Task").WithLocation(24, 19),
+                // (24,19): error CS0656: Missing compiler required member 'B3.Create'
+                //     async T3 f3() => await Task.Delay(3);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(3)").WithArguments("B3", "Create").WithLocation(24, 19),
+                // (25,19): error CS0656: Missing compiler required member 'B4.Task'
+                //     async T4 f4() => await Task.Delay(4);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(4)").WithArguments("B4", "Task").WithLocation(25, 19),
+                // (25,19): error CS0656: Missing compiler required member 'B4.Create'
+                //     async T4 f4() => await Task.Delay(4);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(4)").WithArguments("B4", "Create").WithLocation(25, 19),
+                // (26,19): error CS0656: Missing compiler required member 'B5.Task'
+                //     async T5 f5() => await Task.Delay(5);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(5)").WithArguments("B5", "Task").WithLocation(26, 19),
+                // (26,19): error CS0656: Missing compiler required member 'B5.Create'
+                //     async T5 f5() => await Task.Delay(5);
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "=> await Task.Delay(5)").WithArguments("B5", "Create").WithLocation(26, 19)
+                );
+        }
+
+
+        [Fact]
+        public void AsyncTasklikeBuilderAccessibility()
+        {
+            var source = @"
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+[AsyncBuilder(typeof(B2))] public class T2 { }
+[AsyncBuilder(typeof(B3))] public class T3 { }
+[AsyncBuilder(typeof(B2))] internal class T5 { }
+[AsyncBuilder(typeof(B3))] internal class T6 { }
+
+public class B2 { }
+internal class B3 { }
+public class B5 { }
+internal class B6 { }
```

Comment:
Fixed in https://github.com/dotnet/roslyn/pull/13405/commits/c9633b0dec6321bc78941f4604cae5176927411b